validate.sh

```diff
@@ -40,10 +40,15 @@ else
   test $GOFMT_LINES -eq 0 || die "gofmt needs to be run, ${GOFMT_LINES} files have issues.  Below is a list of files to review:\n`gofmt -l *.go pbs adapters`"
 fi
 
+# Run the actual tests. Make sure there's enough coverage too, if the flags call for it.
 if $COVERAGE; then
   ./scripts/check_coverage.sh
+
+  # Tests take a *really* long time with both the race detector and code coverage, so...
+  # just re-run the tests designed for concurrency again, without code coverage.
+  go test -race $(go list ./... | grep -v /vendor/) -run ^TestRace.*$ -count 2
 else
-  go test $(go list ./... | grep -v /vendor/)
+  go test -race $(go list ./... | grep -v /vendor/) -count 2
```

Comment:
As per our discussion offline, let's see if we can increase the `-- count` value as long as it does not introduce unwanted issues and workarounds. Your idea of using a flag to allow/disallow a race command that has a huge `-- count` number (to enable quick execution) is - superb! Trust you to do what's best!

Reply:
I added a `--race ###` flag so the caller can configure it, and set `travis.yml` to run it 10 times.

I also limited the race detector to only run on tests which target it because I think that filesystem issue I ran into at high values of `--race` is more likely to cause confusion than anything else.